src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -8027,7 +8060,11 @@ namespace WaterThermalTanks {
         Real64 Runtime1 = 0.0;              // Time that heater 1 is running (s)
         Real64 Runtime2 = 0.0;              // Time that heater 2 is running (s)
         bool SetPointRecovered = false;     // Flag to indicate when set point is recovered for the first time
-
+        //Added three variables for desuperheater sourceinlet temperature update
+        Real64 MdotDesuperheaterWater = 0.0;      // mass flow rate of desuperheater source side water, kg/s
+        Real64 DesuperheaterPLR = 0.0;            // Desuperheater part load ratio
+        Real64 DesuperheaterHeaterRate = 0.0;     // Desuperheater heater rate (W)
+        Real64 SourceInletTempSum = 0.0;          // Sum the source inlet temperature in sub time step to calculate average tempearature
```

Comment:
These variables appear to be declared at the top of the function.  Feel free to declare them only within the scope they are used.  In this case, they might be in the right spot, but in general, most of our functions have all the variables declared at the top (Fortran carryover).

Reply:
Okay, understand. In this case, I checked that these declarations were already put in the the smallest scope so I kept them the same.